### PR TITLE
Basic doc phantom types + samples for validations

### PR DESCRIPTION
**Pull request by domdorn**

### CODE REVIEW

File: docs/about/contributing.md [hunk 1]
```diff
@@ -19,7 +19,7 @@ To begin contributing, please follow these steps:
 
 If you don't already have one, sign up for a free [GitHub Account](https://github.com/join?source=header-home).
 
-After you [log into](https://github.com/login) GitHub using your account, go to the [ZIO Project Page](https://github.com/zio/zio), and click on [Fork](https://github.com/zio/zio/fork) to fork the ZIO repository into your own account.
+After you [log into](https://github.com/login) GitHub using your account, go to the [ZIO-Prelude Project Page](https://github.com/zio/zio-prelude), and click on [Fork](https://github.com/zio/zio-prelude/fork) to fork the ZIO-Prelude repository into your own account.
```
Comment (sideeffffect) [inline docs/about/contributing.md:22]: I think that the name of the project is _ZIO Prelude_ (not _ZIO-Prelude_).

File: docs/about/contributing.md [hunk 2]
```diff
@@ -28,12 +28,12 @@ Once you have forked the repository, you can now clone your forked repository to
 To clone your forked repository, first make sure you have installed [Git](https://git-scm.com/downloads), the version control system used by GitHub. Then open a Terminal and type the following commands:
 
 ```bash
-mkdir zio
-cd zio
-git clone git@github.com:your-user-name/zio.git .
+mkdir zio-prelude
```
Comment (sideeffffect) [inline docs/about/contributing.md:31]: AFAIU it, this should stay `zio`. The point is to have a dedicated directory for all repositories under the https://github.com/zio organization.

**Comment by domdorn** [comment]
@sideeffffect please have a look!

**Comment by sideeffffect** [comment]
I can still see a few occurrences of `ZIO-Prelude` :smile_cat:

**Comment by domdorn** [comment]
please check again.

File: docs/overview/phantomtypes.md [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+    import com.acme.myapp.domain._
+    import java.util.UUID
+
+
+    def logSpecific(correlationId: CorrelationId, message: String): Unit = ???
+    // this works
+    logSpecific(CorrelationId.random, "test1"))
+
+    // this doesn't compile, as we're now more strict
+    logSpecific(UUID.randomUUID(), "test2)
```
Comment (sideeffffect) [inline docs/overview/phantomtypes.md:90]: missing `"`

Comment (domdorn) [reply docs/overview/phantomtypes.md]: done

**Review by sideeffffect — APPROVED** [review]

**Comment by sideeffffect** [comment]
Thanks! :heart:

File: docs/about/contributing.md [hunk 4]
```diff
@@ -65,33 +65,49 @@ test:compile
...
 project coreJVM
 ```
 
+### Building the documentation
+If you'd like to contribute some documentation (please please do!), you can build the documentation with SBT using
+```bash
+docusaurusCreateSite
+```
+
+This requires `yarn` to be installed on your machine. Please refer to the [Yarn Installation Page](https://yarnpkg.com/getting-started/install) for instructions on how to install `yarn` on your machine.
```
Comment (phderome) [inline docs/about/contributing.md:80]: that is good! I had to uninstall hadoop yarn!

**Comment by sideeffffect** [comment]
All tests are green, do you want to merge?

**Comment by domdorn** [comment]
@sideeffffect yes please!

**Pull request merged**